internal/sops/hcvault/keysource_test.go

```diff
@@ -15,53 +15,58 @@ import (
 	"time"
 
 	"github.com/hashicorp/vault/api"
+	vaulttransit "github.com/hashicorp/vault/builtin/logical/transit"
+	vaulthttp "github.com/hashicorp/vault/http"
+	"github.com/hashicorp/vault/sdk/logical"
+	"github.com/hashicorp/vault/vault"
 	. "github.com/onsi/gomega"
-	"github.com/ory/dockertest"
 	"go.mozilla.org/sops/v3/hcvault"
 )
 
 var (
-	// testVaultVersion is the version (image tag) of the Vault server image
-	// used to test against.
-	testVaultVersion = "1.10.0"
-	// testVaultToken is the token of the Vault server.
-	testVaultToken = "secret"
 	// testEnginePath is the path to mount the Vault Transit on.
 	testEnginePath = "sops"
 	// testVaultAddress is the HTTP/S address of the Vault server, it is set
 	// by TestMain after booting it.
 	testVaultAddress string
+	// testVaultToken is the token of the Vault server.
+	testVaultToken string
 )
 
 // TestMain initializes a Vault server using Docker, writes the HTTP address to
 // testVaultAddress, waits for it to become ready to serve requests, and enables
 // Vault Transit on the testEnginePath. It then runs all the tests, which can
 // make use of the various `test*` variables.
 func TestMain(m *testing.M) {
-	// Uses a sensible default on Windows (TCP/HTTP) and Linux/MacOS (socket)
-	pool, err := dockertest.NewPool("")
-	if err != nil {
-		logger.Fatalf("could not connect to docker: %s", err)
+	// this is set to prevent "certificate signed by unknown authority" errors
+	os.Setenv("VAULT_INSECURE", "true")
+	t := &testing.T{}
+	coreConfig := &vault.CoreConfig{
+		LogicalBackends: map[string]logical.Factory{
+			"transit": vaulttransit.Factory,
+		},
 	}
-
-	// Pull the image, create a container based on it, and run it
-	resource, err := pool.Run("vault", testVaultVersion, []string{"VAULT_DEV_ROOT_TOKEN_ID=" + testVaultToken})
-	if err != nil {
-		logger.Fatalf("could not start resource: %s", err)
+	cluster := vault.NewTestCluster(t, coreConfig, &vault.TestClusterOptions{
```

Comment:
I'm not sure if this is a problem or not but IIUC we're re-using the cluster created here with the "artificial" testing.T context in tests with a different testing.T. Would this pose any problem w.r.t. log output or anything else?